mantle/system/nproc.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package system
+
+import (
+	"io/ioutil"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/coreos/mantle/system/exec"
+)
+
+// GetProcessors returns a count for number of cores we should use;
+// this value is appropriate to pass to e.g. make -J as well as
+// qemu -smp for example.
+func GetProcessors() (uint, error) {
```

Comment:
nit: `GetProcessors()` seems a bit misleading because we do cap the value at 16. `GetRecommendedProcessorNum()` maybe?

Reply:
another options is to have this function return the correct number of processors and have the caller max it out at 16